internal/services/services.go

```diff
@@ -0,0 +1,601 @@
+	service := DownloadService{}
+	service.SetLocals(c)
+	service.Log = log
+	// Retrieve the injector from the Fiber context
+	reqInjectorI := c.Locals("reqinjector")
+	reqInjector, ok := reqInjectorI.(*do.Injector)
+	if !ok {
+		return nil, fiber.NewError(fiber.StatusInternalServerError, "Failed to retrieve request injector")
+	}
+
+	validator := do.MustInvokeNamed[omnitruck.RequestValidator](reqInjector, "validator")
+	databaseService := do.MustInvokeNamed[dboperations.IDbOperations](reqInjector, "dbService")
+	templateRenderer := do.MustInvokeNamed[template.TemplateRender](reqInjector, "templateRenderer")
+	replicatedService := do.MustInvokeNamed[replicated.IReplicated](reqInjector, "replicated")
+	licenseClient := do.MustInvokeNamed[clients.ILicense](reqInjector, "licenseClient")
+	mode := do.MustInvokeNamed[models.ApiType](reqInjector, "mode")
+	service.Validator = validator
+	service.DatabaseService = databaseService
+	service.TemplateRenderer = templateRenderer
+	service.Replicated = replicatedService
+	service.LicenseClient = licenseClient
+	service.Mode = mode
+	return &service, nil
+}
+func (server *DownloadService) SetLocals(c *fiber.Ctx) {
+	locals := map[string]interface{}{}
+	if c.Locals("valid_license") != nil {
+		requestId := c.Locals("valid_license").(bool)
+		locals["valid_license"] = requestId
+
+	} else {
+		locals["valid_license"] = false
+	}
+
+	if c.Locals("request_id") != nil {
+		requestId := c.Locals("request_id").(string)
+		locals["request_id"] = requestId
+
+	} else {
+		locals["request_id"] = ""
+	}
+	if c.Locals("base_url") != nil {
+		baseUrl := c.Locals("base_url").(string)
+		locals["base_url"] = baseUrl
+	} else {
+		locals["base_url"] = ""
+	}
+	if c.Locals("license_id") != nil {
+		licenseId := c.Locals("license_id").(string)
+		locals["license_id"] = licenseId
+	} else {
+		locals["license_id"] = ""
+	}
+	server.locals = locals
+}
+
+func (server *DownloadService) Omnitruck() *omnitruck.Omnitruck {
+	client := omnitruck.New(server.logCtx())
+
+	return &client
+}
+
+func (server *DownloadService) DynamoServices(db dboperations.IDbOperations) *omnitruck.DynamoServices {
+	service := omnitruck.NewDynamoServices(db, server.logCtx())
+
+	return &service
+}
+
+func (server *DownloadService) PlatformServices() *omnitruck.PlatformServices {
+	service := omnitruck.NewPlatformServices(server.logCtx())
+	return &service
+}
+
+func (server *DownloadService) ReplicatedService(config config.ReplicatedConfig, log logger.Logger) replicated.IReplicated {
+	service := replicated.NewReplicatedImpl(config, log)
+	return service
+}
+
+func (server *DownloadService) logCtx() *log.Entry {
+	return server.Log.WithField("license_id", server.locals["license_id"])
+}
+
+func (server *DownloadService) validLicense() bool {
+	v := server.locals["valid_license"]
+	return v != nil && v.(bool)
+}
+
+func (server *DownloadService) ValidateRequest(params *omnitruck.RequestParams) (string, int, bool) {
```

Comment:
This function should not be part of service. Generally all request coming to server are validated at handler level

Reply:
Done